Modules/Loadable/Terminologies/Logic/vtkSlicerTerminologiesModuleLogic.cxx

```diff
@@ -2558,3 +2561,93 @@ std::vector<std::string> vtkSlicerTerminologiesModuleLogic::FindAnatomicContextN
 
   return foundAnatomicContextNames;
 }
+
+//-----------------------------------------------------------------------------
+bool vtkSlicerTerminologiesModuleLogic::AreSegmentTerminologyEntriesEqual(vtkSegment* segment1, vtkSegment* segment2)
+{
+  if (!segment1 || !segment2)
+  {
+    return false;
+  }
+
+  std::string terminologyEntry1;
+  segment1->GetTag(vtkSegment::GetTerminologyEntryTagName(), terminologyEntry1);
+
+  std::string terminologyEntry2;
+  segment2->GetTag(vtkSegment::GetTerminologyEntryTagName(), terminologyEntry2);
+
+  return this->AreTerminologyEntriesEqual(terminologyEntry1, terminologyEntry2);
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSlicerTerminologiesModuleLogic::AreTerminologyEntriesEqual(std::string terminologyEntry1, std::string terminologyEntry2)
+{
+  vtkNew<vtkSlicerTerminologyEntry> entry1;
+  if (!this->DeserializeTerminologyEntry(terminologyEntry1, entry1))
+  {
+    return false;
```

Comment:
Maybe log error when terminologyEntry is not empty and deserialize fails (because it should not happen)